contrib/windows-cmake/CMakeLists.txt

```
@@ -1,44 +1,62 @@
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.15...3.22)
 project(hwloc
     LANGUAGES C
     VERSION 2.7.0)
 
-# Available configuration options
-# HWLOC_SKIP_LSTOPO doesn't build/install lstopo
-# HWLOC_SKIP_TOOLS doesn't build/install other hwloc tools
-# HWLOC_SKIP_INCLUDES doesn't install headers
+enable_testing()
+
+
+option(HWLOC_ENABLE_TESTING "Enable testing" ON)
+option(HWLOC_SKIP_LSTOPO "don't build/install lstopo")
+option(HWLOC_SKIP_TOOLS "don't build/install other hwloc tools")
+option(HWLOC_SKIP_INCLUDES "don't install headers")
+
+set(TOPDIR ${CMAKE_CURRENT_SOURCE_DIR}/../..)
+
+include(CheckIncludeFile)
+include(CheckSymbolExists)
+include(CheckCSourceCompiles)
+check_include_file("dirent.h" HAVE_DIRENT_H)
+check_include_file("unistd.h" HAVE_UNISTD_H)
+check_symbol_exists(mkstemp "stdlib.h" HAVE_MKSTEMP)
+check_symbol_exists(strncasecmp "strings.h" HAVE_STRNCASECMP)
+if(MSVC AND HAVE_STRNCASECMP)
+    set(hwloc_strncasecmp 1)
+    set(hwloc_strncasecmp_fcn strncasecmp)
+else()
+    set(hwloc_strncasecmp 0)
+    set(hwloc_strncasecmp_fcn strncmp)
+endif()
 
-set(TOPDIR ../..)
+set(SIZEOF_VOID_P ${CMAKE_SIZEOF_VOID_P})
+# disable x86 by default
+set(HWLOC_X86_32_ARCH)
+set(HWLOC_X86_64_ARCH)
```

```suggestion
set(HWLOC_X86_32_ARCH "")
set(HWLOC_X86_64_ARCH "")
```

you want to set them empty and not unset previous values correct?

---

I don't know about the CMake syntax but they should be either 1 or undefined in the end, not empty.

---

Why do you want to have them undefined? Be aware that user might bass -DHWLOC_X86_64_ARCH=1 via the CMake command line. If you don't want that you need to define the variable in local scope.

---

If we change this behavior for CMake, we have to change in configure and in the MSVC solution, and update several code paths using those macros. That's possible, but outside of this PR.

---

Setting them empty in CMake script gives a known state (undefined in header file). Setting them without any argument is akin to boolean false, which is interpreted by `#cmakedefine` statement in the *.h.in files as undefined header.

---

>  Setting them without any argument is akin to boolean false, which is interpreted by #cmakedefine statement in the *.h.in files as undefined header.

Setting them to "" also achieves that. The CMake docs say: 
> "depending on whether VAR is set in CMake to any value not considered a false constant by the if() command."

If docs: 

>"False if the constant is 0, <snip>,  the empty string, or ends in the suffix -NOTFOUND"

@bgoglin:

> If we change this behavior for CMake, we have to change in configure and in the MSVC solution, and update several code paths using those macros. That's possible, but outside of this PR.

I don't want people to pass -DHWLOC_X86_64_ARCH=1 via the cmd line. I just mentioned it that you might need consider that and `set(HWLOC_X86_64_ARCH)` vs `set(HWLOC_X86_32_ARCH "")` have different behavior in this case.